acceptance/tests/facts/schema.rb

```diff
@@ -0,0 +1,117 @@
+test_name "Validate facter output conforms to schema" do
+  tag 'risk:high'
+  confine :except, :platform => 'windows' # See FACT-3479, once resolved this line can be removed
+
+  require 'yaml'
+  require 'ipaddr'
+
+  # Validates passed in output facts correctly conform to the facter schema, facter.yaml.
+  # @param schema_fact The schema fact that matches/corresponds with output_fact
+  # @param schema_fact_value The fact value for the schema fact
+  # @param output_fact The fact that is being validated
+  # @param output_fact The fact value of the output_fact
+  def validate_fact(schema_fact, schema_fact_value, output_fact, output_fact_value)
+    fact_type = output_fact_value.class.to_s.downcase
+    schema_fact_type = schema_fact ? schema_fact_value["type"] : nil
+    fail_test("Fact: #{output_fact} does not exist in schema") unless schema_fact_type
+
+    # For each fact, it is validated by verifying that output_fact_value can
+    # successfully parse to fact_type and the output fact has a matching schema
+    # fact where both their types and name or regex match.
+    if fact_type == "hash"
+      Hash(output_fact_value)
+      fact_type = "map"
+    elsif fact_type == "array"
+      Array(output_fact_value)
+    elsif fact_type == "trueclass" || fact_type == "falseclass"
+      fact_type = "boolean"
+    elsif fact_type == "float"
+      Float(output_fact_value)
+      fact_type = "double"
+    elsif fact_type == "integer"
+      Integer(output_fact_value)
```

Comment:
Since the `output_fact_value` is an instance of `Hash`, `Array`, `Boolean`, `Float`, `Integer` or `String` I don't think we need to validate the `output_fact_value` by creating a copy, e.g. `Integer(...)`.

I do think we want to continue validating IP and mac addresses, since not all `Strings` are valid IP/mac addresses.